app.js

```diff
@@ -38,29 +38,47 @@ var sessionSecret = process.env.SESSION_SECRET || settings.secret;
 var db = mongoose.connection;
 var dbOptions = { server: { socketOptions: { keepAlive: 1 } } };
 
+var fs = require('fs');
+var http = require('http');
+var https = require('https');
+var sslOptions = null;
+var server = http.createServer(app);
+var secureServer = null;
+
 app.set('port', process.env.PORT || 8080);
+app.set('securePort', process.env.SECURE_PORT || null);
 
 // Connect to the database
 mongoose.connect(connectStr, dbOptions);
 db.on('error', console.error.bind(console, 'connection error:'));
-db.once('open', function () {
-  app.listen(app.get('port'));
-});
+db.once('open', function () {});
 
 var sessionStore = new MongoStore({ mongooseConnection: db });
 
 // Force HTTPS
-if (app.get('port') === 443) {
+if (app.get('securePort')) {
```

Comment:
If you don't set the SECURE_PORT environment variable then it runs using
http on PORT.
On May 29, 2015 3:48 AM, "Marti Martz" <notifications@github.com> wrote:

> In app.js
> <https://github.com/OpenUserJs/OpenUserJS.org/pull/635#discussion_r31308229>
> :
>
> >
> >  var sessionStore = new MongoStore({ mongooseConnection: db });
> >
> >  // Force HTTPS
> > -if (app.get('port') === 443) {
> > +if (app.get('securePort')) {
>
> @sizzlemctwizzle <https://github.com/sizzlemctwizzle>
> So you are doing this programmatically now from the environment now
> instead of what was done at 932d1c9
> <https://github.com/OpenUserJs/OpenUserJS.org/commit/932d1c906d70d6b305231d1ad381f60a7a717ea3>
> ? e.g. which no longer allows http://openuserjs.org ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenUserJs/OpenUserJS.org/pull/635/files#r31308229>.
>